packages/web-wallet/src/contexts/hooks/useWalletConnection.ts

```diff
@@ -178,191 +192,235 @@ export function useWalletConnection(options: UseWalletConnectionOptions): Wallet
     window.location.reload();
   };
 
-  const checkExistingConnection = async (signal?: AbortSignal) => {
-    if (isCheckingRef.current) {
-      return;
+  /**
+   * Verifies snap installation and permissions.
+   * Throws typed errors for different snap states.
+   */
+  const verifySnapInstallation = async (): Promise<void> => {
+    // wallet_getSnaps is a MetaMask wallet method, not a snap method
+    // So we need to call it via window.ethereum directly
+    const snapsResponse = await window.ethereum.request({
```

Comment:
This RPC calls metamask directly, not the snap, that's why I'm using `window.ethereum`